subsys/sdfw_services/services/device_info/device_info_service.c

```diff
@@ -0,0 +1,151 @@
+			  cbor_decode_device_info_resp);
+
+static int get_device_info_data(struct device_info *device_info_data)
+{
+	int ret = -SSF_ENODATA;
+
+	if (device_info_data != NULL) {
+		const uint8_t *rsp_pkt;
+
+		int stat = stat_INTERNAL_ERROR_c;
+
+		struct device_info_req device_info_request = {
+			.device_info_req_msg.read_req_action.operation_choice =
+				operation_READ_DEVICE_INFO_c,
+		};
+
+		struct read_resp response_message = {0};
+
+		struct device_info_resp device_info_response = {
+			.device_info_resp_msg = response_message,
+		};
+
+		ret = ssf_client_send_request(&device_info_srvc, &device_info_request,
+					      &device_info_response, &rsp_pkt);
+		if (ret == 0) {
+			if (device_info_response.device_info_resp_msg.read_resp_action
+				    .operation_choice == operation_READ_DEVICE_INFO_c) {
+				stat = device_info_response.device_info_resp_msg.read_resp_status
+					       .stat_choice;
+				if (stat == stat_SUCCESS_c) {
+
+					/* copy device info data to provided struction */
+					device_info_data->device_info_partno =
+						device_info_response.device_info_resp_msg
+							.read_resp_data.device_info_partno;
+
+					device_info_data->device_info_type =
+						device_info_response.device_info_resp_msg
+							.read_resp_data.device_info_type;
+
+					device_info_data->device_info_hwrevision =
+						device_info_response.device_info_resp_msg
+							.read_resp_data.device_info_hwrevision;
+
+					device_info_data->device_info_productionrevision =
+						device_info_response.device_info_resp_msg
+							.read_resp_data
+							.device_info_productionrevision;
+
+					memcpy((uint8_t *)(device_info_data->device_info_testimprint
+								   .value),
+					       (uint8_t *)(device_info_response.device_info_resp_msg
+								   .read_resp_data
+								   .device_info_testimprint.value),
+					       device_info_response.device_info_resp_msg
+						       .read_resp_data.device_info_testimprint.len);
+
+					device_info_data->device_info_testimprint.len =
+						device_info_response.device_info_resp_msg
+							.read_resp_data.device_info_testimprint.len;
+
+					memcpy((uint8_t *)(device_info_data->device_info_uuid
+								   .value),
+					       (uint8_t *)(device_info_response.device_info_resp_msg
+								   .read_resp_data.device_info_uuid
+								   .value),
+					       device_info_response.device_info_resp_msg
+						       .read_resp_data.device_info_uuid.len);
+
+					device_info_data->device_info_uuid.len =
+						device_info_response.device_info_resp_msg
+							.read_resp_data.device_info_uuid.len;
+
+					/* operation is successful */
+					ret = 0;
+				} else {
+					/* operation failed on server side */
+					ret = -SSF_EPROTO;
+				}
+			} else {
+				/* the received response message is not a read device info response
+				 */
+				ret = -SSF_EBADMSG;
+			}
+
+			/* at least a response message was received, free response packet buffer */
+			ssf_client_decode_done(rsp_pkt);
```

Comment:
@anhmolt is it OK to only call this here?

Reply:
Yes, this looks good